Gems/O3DE/GeomNodes/Code/Source/Editor/Rendering/GNModelData.cpp

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Editor/Rendering/GNModelData.h>
+#include <Editor/Systems/GNParamContext.h>
+#include <AzCore/std/containers/set.h>
+#include <AzCore/std/containers/map.h>
+#include <Editor/Common/GNAPI.h>
+
+namespace GeomNodes
+{
+    GNModelData::GNModelData()
+    {
+    }
+
+    GNModelData::GNModelData(AZ::u64 mapId)
+    {
+		ReadData(mapId);
+    }
+
+    void GNModelData::ReadData(AZ::u64 mapId)
+    {
+		m_meshes.clear();
+		
+		if (API::OpenSHM(mapId))
```

Comment:
O3DE uses 4 spaces for indentation, this appears to have 8

Reply:
Right. This might be caused by Visual Studio and Visual Assist formatting. But will definitely fix this.

Reply:
fixed in commit [d756b38](https://github.com/o3de/o3de-technicalart/pull/11/commits/d756b38541172fd3e0051140ca96cc9bd8b05388)